### ISSUE TITLE
Using other framework instead of Gorilla Mux

### ISSUE THREAD

**Issue by yuvicc** [jenkins-x/jx-cdevents-adapter#13]
### Is your proposal related to a problem?
We're using Gorilla Mux as http webframework which has very low performance. We can see [here](https://www.techempower.com/benchmarks/#section=data-r21&test=composite).
Also Gorilla Mux project has been archived due to no active maintainer, which means there will be no future updates. 

### Describe the solution you'd like
I'ld like to suggest the use of [Fiber](https://gofiber.io/) which is very fast and has very low latency. The fiber can be easily understood by the new developers, also it has nice documentation. The composite benchmarks of fiber is far above than the gorilla/mux framework.

**Comment by ankitm123** [comment]
Why not use something that is compatible with net/http? How about chi which is a pretty performant router and actively maintained. 
Fiber is good for very specific use cases, but the fact that it breaks compatibility with standard library makes it pretty unattractive to me.

Also if performance is an issue, we should measure it first in this repo before we seriously think about fiber.

**Comment by yuvicc** [comment]
Yes, I think you're right. Since we're building a restful type API, chi would be better for building a scalable RESTful API and it is also compatible for net/http as well as standard library.

Should I go forward with chi?

**Comment by ankitm123** [comment]
Yes, that would be awesome, thanks for looking into this :)
